src/ConfigManager.cpp

```diff
@@ -20,64 +20,73 @@ void ConfigManager::setup() {
   char ssid[SSID_LENGTH];
   char password[PASSWORD_LENGTH];
 
-  DebugPrintln(F("Reading saved configuration"));
-
   DebugPrint(F("MAC: "));
   DebugPrintln(WiFi.macAddress());
 
+  DebugPrintln(F("Reading saved configuration"));
   EEPROM.get(0, magic);
-  EEPROM.get(MAGIC_LENGTH, ssid);
-  DebugPrint(F("SSID: \""));
-  DebugPrint(ssid);
-  DebugPrintln(F("\""));
-  EEPROM.get(MAGIC_LENGTH + SSID_LENGTH, password);
-  readConfig();
 
   if (memcmp(magic, magicBytes, MAGIC_LENGTH) == 0) {
+    EEPROM.get(MAGIC_LENGTH, ssid);
+    DebugPrint(F("SSID: \""));
+    DebugPrint(ssid);
+    DebugPrintln(F("\""));
+
+    EEPROM.get(MAGIC_LENGTH + SSID_LENGTH, password);
+    readConfig();
+
     WiFi.begin(ssid, password[0] == '\0' ? NULL : password);
+
     if (wifiConnected()) {
       DebugPrint(F("Connected to "));
       DebugPrint(ssid);
       DebugPrint(F(" with "));
       DebugPrintln(WiFi.localIP());
 
       WiFi.mode(WIFI_STA);
+
       startApi();
-      return;
```

Comment:
This doesnt seem right. It means we will fall through and `startAP`.

Reply:
guard against that by checking we are not in `STATION` mode

```
if (this->getMode() != station) {
    startAP();
    startAPApi();
  }
```